argo-rollouts-demo/README.md

```diff
@@ -0,0 +1,229 @@
+- [Install](https://kubernetes.io/docs/tasks/tools/#kubectl) `kubectl`
+- [Install](https://kubectl.docs.kubernetes.io/installation/kustomize/) `kustomize`
+
+## Setup the Pixie Metrics Server
+
+1. Clone this repo and navigate to the `argo-rollouts-demo` folder:
+
+```
+git clone https://github.com/pixie-io/pixie-demos.git
+cd pixie-demos/argo-rollouts-demo
+```
+
+2. Create a secret containing the Pixie API credentials for your Kubernetes cluster:
+
+```
+# Get your current cluster name from your Kubernetes context.
+kubectl config current-context
+
+# Get the Pixie Cluster ID for the above cluster name.
+# Record the value of the `ID` column for this cluster.
+px get viziers
+
+# Create an API key. Record the value of the `Key` parameter.
+px api-key create
+
+# Create `px-metrics` namespace.
+kubectl create namespace px-metrics
+
+# Create the secret.
+kubectl -n px-metrics create secret generic px-credentials --from-literal=px-api-key=<YOUR API KEY VALUE HERE> --from-literal=px-cluster-id=<YOUR CLUSTER ID VALUE HERE>
+```
+
+3. Install the Pixie metrics server onto your cluster with:
+
+```
+kubectl apply -f px-metrics.yaml
+```
+
+## Setup the Demo Application
+
+1. [Install](https://argoproj.github.io/argo-rollouts/installation/#controller-installation) Argo Rollouts onto your cluster with:
+
+```
+kubectl create namespace argo-rollouts
+kubectl apply -n argo-rollouts -f https://github.com/argoproj/argo-rollouts/releases/latest/download/install.yaml
+```
+
+2. [Install](https://argoproj.github.io/argo-rollouts/installation/#kubectl-plugin-installation) the Argo Rollouts `kubectl` plugin. To use Homebrew, run:
+
+```
+brew install argoproj/tap/kubectl-argo-rollouts
+
+```
+
+2. Apply the manifests (including the application [Rollout](https://github.com/pixie-io/pixie-demos/tree/main/argo-rollouts-demo/canary/rollout-with-analysis.yaml) and [AnalysisTemplate](https://github.com/pixie-io/pixie-demos/tree/main/argo-rollouts-demo/canary/pixie-analysis.yaml)):
+
+```
+kustomize build . | kubectl apply -f -
+```
+
+3. Get the `EXTERNAL-IP` address for the `canary-demo-preview` service:
+
+```
+kubectl get svc canary-demo-preview
+```
+
+4. Navigate to the External IP in your browser to access the demo application front-end. Leave this open.
+
+<br>
+<img src=".readme_assets/front-end.png"  width="550" align="right">
+<br>
+
+Each of these colorful squares is a request the browser is making to the backend.
+
+The backend responds with a color that indicates which version of the app it is. In this case we're getting blue boxes for the application image with the `blue` tag set in the [Rollout](https://github.com/pixie-io/pixie-demos/tree/main/argo-rollouts-demo/canary/rollout-with-analysis.yaml) yaml.
+
+The bar chart at the bottom represents the percentage of requests that were handled by the different backends (stable, canary). We will see this in action in a minute.
```

Comment:
"The bar chart at the bottom represents the percentage of requests that were handled by the different backends (stable, canary). **Currently you should see all requests are handled by the "stable" backend.**  We will see this in action in a minute."